common/evidenceextractor_plugin.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2021 Contributors to the Veraison project.
+// SPDX-License-Identifier: Apache-2.0
+
+package common
+
+import (
+	"encoding/json"
+	"log"
+	"net/rpc"
+
+	"github.com/hashicorp/go-plugin"
+)
+
+type EvidenceExtractorPlugin struct {
+	Impl IEvidenceExtractor
+}
+
+func (p EvidenceExtractorPlugin) Server(*plugin.MuxBroker) (interface{}, error) {
+	return &EvidenceExtractorServer{Impl: p.Impl}, nil
+}
+
+func (p EvidenceExtractorPlugin) Client(b *plugin.MuxBroker, c *rpc.Client) (interface{}, error) {
+	return &EvidenceExtractorRPC{client: c}, nil
+}
+
+type EvidenceExtractorServer struct {
+	Impl IEvidenceExtractor
+}
+
+func (s EvidenceExtractorServer) GetName(args interface{}, resp *string) error {
+	*resp = s.Impl.GetName()
+	return nil
```

Comment:
If I am not mistaken, GetName can return a "" string which is an Error, however here without checking for null string, we are returning a nil which means no error?

Reply:
Here we're just forwarding the result form the implementation `GetName`. Since calling the underlying implementation doesn't result in an error, this shouldn't either. It is part of the definition of `GetName` that it does not error. If `""` is returned it will simply be forwarded to the calling code, as if the calling code invoked `Impl.GetName()` directly. It would then be up to the calling code to handle this case.

Reply:
Thanks for the clarification, please resolve the comment then !